### DISCUSSION TITLE
Challenge #5: classic FizzBuzz

### DISCUSSION THREAD

**Discussion by CosmicKetchup** [TestPlant/sensetalk-solutions#26] [Accepted Challenges]
> To most developers, **FizzBuzz** is nothing more than a trivial formality often utilized within interviews to quickly identify those that have basic programming knowledge from those that do not.

#### Automation Requirement(s)
- write a command that does the following:
  - print each number starting from `1` through to `100` on a new line
  - if the number is a multiple of `3`, print the string `Fizz` instead of the number
  - if the number is a multiple of `5`, print the string `Buzz` instead of the number
  - if the number is a multiple of **both** `3` _and_ `5`, print the string `FizzBuzz` instead of the number

#### Notes

**Comment by DeepPurpleActual** [comment]
If anyone wants to check the syntax of the portion that I commented out, I would like your feedback.

**Comment by DeepPurpleActual** [comment]
I am curious why this multi-case statement failed.

```
If myNumber ...
		... is a multiple of 15 then Set myFB_Solution to "FizzBuzz"
		... is a multiple of 5 then Set myFB_Solution to "Buzz"
		... is a multiple of 3 then Set myFB_Solution to "Fizz"
End If
```

**Reply by CosmicKetchup** [reply]
I originally thought this had to do with the limited number of comparison operators supported by this syntax, but Doug looked into it more:

> That’s a good guess — in fact that’s what I thought at first, too. But the limited set of operators only applies to those you can use when you include the operator on the first line (like if `myVar contains…` or whatever).

> In this case, I see that the problem is a mistake I made in the grammar for the SenseTalk compiler that leaves out the higher-precedence operators. Basically, if you look at the precedence table [here](http://docs.eggplantsoftware.com/ePF/SenseTalk/stk-expressions.htm) everything that’s shown as 1st through 7th priority (and also, oddly, 11th priority) can not be used in a multi-case if, which unfortunately includes `is divisible by`. That wasn’t intentional, so please file a bug report and I should be able to fix it in the future.
